File: UI/src/app/shared/models/reviews.model.ts
export interface ReviewsModel {
    productId?: number;
    averageRating: number;
    reviews?: ReviewInfoModel[];
}

export interface ReviewInfoModel {
    reviewId: number;
    customerId?: number;
    userName?: string;
    rating?: number;
    comment?: string;
    createdAt?: Date;
}

export interface ReviewInsertModel {
    customerId: number;
    productId: number;
    rating: number;
    comment?: string;
}

File: API/Models/ReviewsModel.cs
namespace ModelService
{
    public class ReviewsModel
    {
        public int? ProductId { get; set; }
        public decimal AverageRating { get; set; }
        public List<ReviewInfoModel>? Reviews { get; set; }
    }

    public class ReviewInfoModel
    {
        public int ReviewId { get; set; }
        public int? CustomerId { get; set; }
        public string? UserName { get; set; }
        public int? Rating { get; set; }
        public string? Comment { get; set; }
        public DateTime? CreatedAt { get; set; }
    }

    public class ReviewInsertModel
    {
        public int CustomerId { get; set; }
        public int ProductId { get; set; }
        public int Rating { get; set; }
        public string? Comment { get; set; }
    }
}